src/interactivity/commonFeature.js

```diff
@@ -0,0 +1,45 @@
+
+function _getGeomPropertiesFromPieces (pieces) {
+    const geomPropertiesFromPieces = pieces.map(piece => {
+        const [x, y] = piece.getCentroid();
+        const aabb = piece._rawFeature._dataframe._aabb[piece._rawFeature._index];
+        return { centroid: { x, y }, aabb };
+    });
+
+    const geomProperties = geomPropertiesFromPieces.filter(piece => {
+        const { x, y } = piece.centroid;
+        return !(isNaN(x) || isNaN(y));
+    });
+
+    return geomProperties;
+}
+
+/**
+ * Get a 'compound Feature'. It is a Feature created from a set of 'feature' pieces
+ * @param {Array} featurePieces
+ */
+export function getCompoundFeature (featurePieces) {
+    const exemplar = featurePieces[0];
+    if (featurePieces.length === 1) return exemplar;
+
+    const geomProperties = _getGeomPropertiesFromPieces(featurePieces);
+    Object.defineProperty(exemplar, 'getCentroid', {
```

Comment:
What's the reason to use a getter here?

Reply:
I was trying to gain some performance here, by lazy calculating some stuff just when invoking the method, the same as when creating a Feature from scratch